5 lab(mvvm_verification ): реализовал лабораторную 5 по предмету ООАиП.

RockInHead opened the pull request.

RockInHead committed:
Init(View): установил CommunityToolkit.Mvvm.

RockInHead committed:
refactoring(View): закомментировал свою реализацию комманд и некоторых свойств. Использовал готовую из Community MVVM Toolkit.

RockInHead committed:
features(Contacts): разделил проект на 3 части: View, Model и ViewModel. Удалил лишнее.

sandra155464 commented on Programming/Contacts/ViewModel/MainVM.cs:
```diff
@@ -182,58 +162,66 @@ public void ApplyContact(object parameter)
     /// </summary>
-    /// <param name="propertyName">Имя измененного свойства.</param>
-    protected void OnPropertyChanged(string propertyName)
+    private void CancelEdit()
```
Этот метод нигде не используется - может, его стоит убрать?

sandra155464 commented on Programming/Contacts/Contacts/View.csproj:
```diff
@@ -14,9 +14,15 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="CommunityToolkit.Mvvm" Version="8.4.0" />
```
У проектов View и ViewModel есть подключенный NuGet пакет Newtonsoft.Json. Он точно нужен этим проектам?

sandra155464 commented on Programming/Contacts/Model/Model.csproj:
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
```
В методичке было сказано: "Если классы уровня Model реализуют интерфейс INotifyPropertyChanged, IDataErrorInfo или иные интерфейсы MVVM, то данный проект также может подключить пакет MVVM Toolkit", т.е. класс Contact нужно отнаследовать от ObservableObject аналогично классу MainVM

sandra155464 replied:
Также в классе Contact 168-ая строка превышает по длине 100 символов, поэтому сравнение "разделить" (аналогично тому, как сделано на строках 156-158). Также в проверке на наличие в строке символа '@' необходимо заменить string на char, т.е. писать с использованием одинарных кавычек:
``` cs
!Email.Contains('@')
```

sandra155464 commented:
Оставила замечания по лабораторной

sandra155464 commented:
На всякий случай, уточню - это ещё не означает, что лабораторная принята или что-то наподобие, просто у пул-реквеста изначально был статус "Draft", а кнопочкой "Ready for review" я перевела его в статус "Open", потому что "Draft" нельзя мёрджить
![image](https://github.com/user-attachments/assets/5701c072-af63-4bce-8918-4f176dbb74d6)

RockInHead committed:
refactoring(Contacts): исправил ошибки.

- убрал неиспользуемую библиотеку Newtonsoft.Json в View и ViewModel.
- удалил неиспользуемый метод CancelEdit() в MainVM.cs
- отнаследовал класс Contact.cs от ObservableObject и упростил класс, убрав свойства.
